components/Carbon.js

```diff
@@ -402,27 +409,34 @@ function selectedLinesReducer({ prevLine, selected }, { type, lineNumber, numLin
   }
 }
 
-function useGutterClickHandler(props) {
-  const editorRef = React.useRef(null)
+function useSelectedLines(props, editorRef) {
   const [state, dispatch] = React.useReducer(selectedLinesReducer, {
     prevLine: null,
     selected: {}
   })
 
   React.useEffect(() => {
-    if (editorRef.current) {
-      editorRef.current.display.view.forEach((line, i) => {
-        if (line.text && line.gutter) {
-          line.text.style.opacity = state.selected[i] === false ? 0.5 : 1
-          line.gutter.style.opacity = state.selected[i] === false ? 0.5 : 1
+    if (editorRef.current && Object.keys(state.selected).length > 0) {
+      editorRef.current.editor.display.view.forEach((line, i) => {
+        if (line.text) {
+          line.text.style.opacity = state.selected[i] === true ? 1 : 0.5
+        }
+        if (line.gutter) {
+          line.gutter.style.opacity = state.selected[i] === true ? 1 : 0.5
         }
       })
     }
-  }, [state.selected, props.children, props.config])
+  }, [state.selected, props.children, props.config, editorRef])
 
-  return React.useCallback(function onGutterClick(editor, lineNumber, gutter, e) {
-    editorRef.current = editor
+  React.useEffect(() => {
+    let selectedLines = (props.config.selectedLines || '')
+      .split(',')
+      .map(i => parseInt(i))
+      .filter(i => !isNaN(i))
```

Comment:
This number filtering/mapping should happen during the query param parsing here: https://github.com/carbon-app/carbon/pull/878/files#diff-7bd084e1642aa17e0074c18f5a25a0aeR76

Reply:
sure, I will move it